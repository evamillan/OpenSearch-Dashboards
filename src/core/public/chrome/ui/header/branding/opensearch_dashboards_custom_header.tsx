/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import { EuiHeader, EuiHeaderProps } from '@elastic/eui';
import React from 'react';
import styled from 'styled-components';
import { ChromeBranding } from '../../../chrome_service';

interface StyledHeaderProps {
  $branding: ChromeBranding;
}

export type CustomHeaderProps = EuiHeaderProps & {
  branding: ChromeBranding;
};

const StyledHeader = styled(EuiHeader)<StyledHeaderProps>`
  background-color: ${(props) => props.$branding.colors?.headerBackgroundColor};
  border-bottom-color: ${(props) => props.$branding.colors?.headerBackgroundColor};
  .euiHeaderLink,
  .euiHeaderSectionItemButton {
    color: ${(props) => props.$branding.colors?.headerLinkColor};
  }
`;

export const CustomHeader = (props: CustomHeaderProps) => {
  const { branding, ...rest } = props;

  return <StyledHeader $branding={branding} {...rest} />;
};
