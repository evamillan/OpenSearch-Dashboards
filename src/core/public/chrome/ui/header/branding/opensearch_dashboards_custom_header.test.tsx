/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { mountWithIntl } from 'test_utils/enzyme_helpers';
import { CustomHeader } from './opensearch_dashboards_custom_header';

describe('Styled header', () => {
  it('rendered using default colors', () => {
    const props = {
      branding: { colors: {} },
    };
    const component = mountWithIntl(<CustomHeader {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('rendered using custom colors', () => {
    const props = {
      branding: {
        colors: {
          headerBackgroundColor: '#000000',
          headerLinkColor: '#ffffff',
        },
      },
    };
    const component = mountWithIntl(<CustomHeader {...props} />);
    expect(component).toMatchSnapshot();
  });
});
