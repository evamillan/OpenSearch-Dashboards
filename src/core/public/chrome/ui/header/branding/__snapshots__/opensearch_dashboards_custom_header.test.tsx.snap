// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled header rendered using custom colors 1`] = `
<CustomHeader
  branding={
    Object {
      "colors": Object {
        "headerBackgroundColor": "#000000",
        "headerLinkColor": "#ffffff",
      },
    }
  }
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
>
  <Styled(EuiHeader)
    $branding={
      Object {
        "colors": Object {
          "headerBackgroundColor": "#000000",
          "headerLinkColor": "#ffffff",
        },
      }
    }
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {
          "date": Object {
            "full": Object {
              "day": "numeric",
              "month": "long",
              "weekday": "long",
              "year": "numeric",
            },
            "long": Object {
              "day": "numeric",
              "month": "long",
              "year": "numeric",
            },
            "medium": Object {
              "day": "numeric",
              "month": "short",
              "year": "numeric",
            },
            "short": Object {
              "day": "numeric",
              "month": "numeric",
              "year": "2-digit",
            },
          },
          "number": Object {
            "currency": Object {
              "style": "currency",
            },
            "percent": Object {
              "style": "percent",
            },
          },
          "relative": Object {
            "days": Object {
              "units": "day",
            },
            "hours": Object {
              "units": "hour",
            },
            "minutes": Object {
              "units": "minute",
            },
            "months": Object {
              "units": "month",
            },
            "seconds": Object {
              "units": "second",
            },
            "years": Object {
              "units": "year",
            },
          },
          "time": Object {
            "full": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
              "timeZoneName": "short",
            },
            "long": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
              "timeZoneName": "short",
            },
            "medium": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
            },
            "short": Object {
              "hour": "numeric",
              "minute": "numeric",
            },
          },
        },
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {},
        "now": [Function],
        "onError": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": null,
      }
    }
  >
    <EuiHeader
      className="sc-bczRLJ hHcqzY"
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {
            "date": Object {
              "full": Object {
                "day": "numeric",
                "month": "long",
                "weekday": "long",
                "year": "numeric",
              },
              "long": Object {
                "day": "numeric",
                "month": "long",
                "year": "numeric",
              },
              "medium": Object {
                "day": "numeric",
                "month": "short",
                "year": "numeric",
              },
              "short": Object {
                "day": "numeric",
                "month": "numeric",
                "year": "2-digit",
              },
            },
            "number": Object {
              "currency": Object {
                "style": "currency",
              },
              "percent": Object {
                "style": "percent",
              },
            },
            "relative": Object {
              "days": Object {
                "units": "day",
              },
              "hours": Object {
                "units": "hour",
              },
              "minutes": Object {
                "units": "minute",
              },
              "months": Object {
                "units": "month",
              },
              "seconds": Object {
                "units": "second",
              },
              "years": Object {
                "units": "year",
              },
            },
            "time": Object {
              "full": Object {
                "hour": "numeric",
                "minute": "numeric",
                "second": "numeric",
                "timeZoneName": "short",
              },
              "long": Object {
                "hour": "numeric",
                "minute": "numeric",
                "second": "numeric",
                "timeZoneName": "short",
              },
              "medium": Object {
                "hour": "numeric",
                "minute": "numeric",
                "second": "numeric",
              },
              "short": Object {
                "hour": "numeric",
                "minute": "numeric",
              },
            },
          },
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {},
          "now": [Function],
          "onError": [Function],
          "textComponent": Symbol(react.fragment),
          "timeZone": null,
        }
      }
    >
      <div
        className="euiHeader euiHeader--default euiHeader--static sc-bczRLJ hHcqzY"
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": Object {
              "date": Object {
                "full": Object {
                  "day": "numeric",
                  "month": "long",
                  "weekday": "long",
                  "year": "numeric",
                },
                "long": Object {
                  "day": "numeric",
                  "month": "long",
                  "year": "numeric",
                },
                "medium": Object {
                  "day": "numeric",
                  "month": "short",
                  "year": "numeric",
                },
                "short": Object {
                  "day": "numeric",
                  "month": "numeric",
                  "year": "2-digit",
                },
              },
              "number": Object {
                "currency": Object {
                  "style": "currency",
                },
                "percent": Object {
                  "style": "percent",
                },
              },
              "relative": Object {
                "days": Object {
                  "units": "day",
                },
                "hours": Object {
                  "units": "hour",
                },
                "minutes": Object {
                  "units": "minute",
                },
                "months": Object {
                  "units": "month",
                },
                "seconds": Object {
                  "units": "second",
                },
                "years": Object {
                  "units": "year",
                },
              },
              "time": Object {
                "full": Object {
                  "hour": "numeric",
                  "minute": "numeric",
                  "second": "numeric",
                  "timeZoneName": "short",
                },
                "long": Object {
                  "hour": "numeric",
                  "minute": "numeric",
                  "second": "numeric",
                  "timeZoneName": "short",
                },
                "medium": Object {
                  "hour": "numeric",
                  "minute": "numeric",
                  "second": "numeric",
                },
                "short": Object {
                  "hour": "numeric",
                  "minute": "numeric",
                },
              },
            },
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "en",
            "messages": Object {},
            "now": [Function],
            "onError": [Function],
            "textComponent": Symbol(react.fragment),
            "timeZone": null,
          }
        }
      />
    </EuiHeader>
  </Styled(EuiHeader)>
</CustomHeader>
`;

exports[`Styled header rendered using default colors 1`] = `
<CustomHeader
  branding={
    Object {
      "colors": Object {},
    }
  }
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
>
  <Styled(EuiHeader)
    $branding={
      Object {
        "colors": Object {},
      }
    }
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {
          "date": Object {
            "full": Object {
              "day": "numeric",
              "month": "long",
              "weekday": "long",
              "year": "numeric",
            },
            "long": Object {
              "day": "numeric",
              "month": "long",
              "year": "numeric",
            },
            "medium": Object {
              "day": "numeric",
              "month": "short",
              "year": "numeric",
            },
            "short": Object {
              "day": "numeric",
              "month": "numeric",
              "year": "2-digit",
            },
          },
          "number": Object {
            "currency": Object {
              "style": "currency",
            },
            "percent": Object {
              "style": "percent",
            },
          },
          "relative": Object {
            "days": Object {
              "units": "day",
            },
            "hours": Object {
              "units": "hour",
            },
            "minutes": Object {
              "units": "minute",
            },
            "months": Object {
              "units": "month",
            },
            "seconds": Object {
              "units": "second",
            },
            "years": Object {
              "units": "year",
            },
          },
          "time": Object {
            "full": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
              "timeZoneName": "short",
            },
            "long": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
              "timeZoneName": "short",
            },
            "medium": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
            },
            "short": Object {
              "hour": "numeric",
              "minute": "numeric",
            },
          },
        },
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {},
        "now": [Function],
        "onError": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": null,
      }
    }
  >
    <EuiHeader
      className="sc-bczRLJ lnGSGL"
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {
            "date": Object {
              "full": Object {
                "day": "numeric",
                "month": "long",
                "weekday": "long",
                "year": "numeric",
              },
              "long": Object {
                "day": "numeric",
                "month": "long",
                "year": "numeric",
              },
              "medium": Object {
                "day": "numeric",
                "month": "short",
                "year": "numeric",
              },
              "short": Object {
                "day": "numeric",
                "month": "numeric",
                "year": "2-digit",
              },
            },
            "number": Object {
              "currency": Object {
                "style": "currency",
              },
              "percent": Object {
                "style": "percent",
              },
            },
            "relative": Object {
              "days": Object {
                "units": "day",
              },
              "hours": Object {
                "units": "hour",
              },
              "minutes": Object {
                "units": "minute",
              },
              "months": Object {
                "units": "month",
              },
              "seconds": Object {
                "units": "second",
              },
              "years": Object {
                "units": "year",
              },
            },
            "time": Object {
              "full": Object {
                "hour": "numeric",
                "minute": "numeric",
                "second": "numeric",
                "timeZoneName": "short",
              },
              "long": Object {
                "hour": "numeric",
                "minute": "numeric",
                "second": "numeric",
                "timeZoneName": "short",
              },
              "medium": Object {
                "hour": "numeric",
                "minute": "numeric",
                "second": "numeric",
              },
              "short": Object {
                "hour": "numeric",
                "minute": "numeric",
              },
            },
          },
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {},
          "now": [Function],
          "onError": [Function],
          "textComponent": Symbol(react.fragment),
          "timeZone": null,
        }
      }
    >
      <div
        className="euiHeader euiHeader--default euiHeader--static sc-bczRLJ lnGSGL"
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": Object {
              "date": Object {
                "full": Object {
                  "day": "numeric",
                  "month": "long",
                  "weekday": "long",
                  "year": "numeric",
                },
                "long": Object {
                  "day": "numeric",
                  "month": "long",
                  "year": "numeric",
                },
                "medium": Object {
                  "day": "numeric",
                  "month": "short",
                  "year": "numeric",
                },
                "short": Object {
                  "day": "numeric",
                  "month": "numeric",
                  "year": "2-digit",
                },
              },
              "number": Object {
                "currency": Object {
                  "style": "currency",
                },
                "percent": Object {
                  "style": "percent",
                },
              },
              "relative": Object {
                "days": Object {
                  "units": "day",
                },
                "hours": Object {
                  "units": "hour",
                },
                "minutes": Object {
                  "units": "minute",
                },
                "months": Object {
                  "units": "month",
                },
                "seconds": Object {
                  "units": "second",
                },
                "years": Object {
                  "units": "year",
                },
              },
              "time": Object {
                "full": Object {
                  "hour": "numeric",
                  "minute": "numeric",
                  "second": "numeric",
                  "timeZoneName": "short",
                },
                "long": Object {
                  "hour": "numeric",
                  "minute": "numeric",
                  "second": "numeric",
                  "timeZoneName": "short",
                },
                "medium": Object {
                  "hour": "numeric",
                  "minute": "numeric",
                  "second": "numeric",
                },
                "short": Object {
                  "hour": "numeric",
                  "minute": "numeric",
                },
              },
            },
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "en",
            "messages": Object {},
            "now": [Function],
            "onError": [Function],
            "textComponent": Symbol(react.fragment),
            "timeZone": null,
          }
        }
      />
    </EuiHeader>
  </Styled(EuiHeader)>
</CustomHeader>
`;
